src/component/Calendar/components/CustomHeader.js

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import styled from "styled-components";
+
+export default function CustomHeader({ date, setDate, month, handleMonthChange }) {
+  return (
+    <Wrapper>
+        <ColorBox>
+            <span className='today'>Today</span>
+            <span className='great'>Great</span>
+            <span className='sorry'>Sorry</span>
+        </ColorBox>
+        <Title>
+            <h2>To Do Calendar</h2>
+            <span>({date.toISOString().split('T')[0]})</span>
+        </Title>
+        <SelectMonth value={month} onChange={handleMonthChange}>
+            {Array.from({ length: 12 }, (_, i) => (
+                <option key={i} value={i}>
+                {new Date(0, i).toLocaleString('en-US', { month: 'long' })} {/* 월 이름 표시 */}
+                </option>
+            ))}
+        </SelectMonth>
+    </Wrapper>
+  );
+}
+const Wrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 0 20px;
+  align-items: center;
+  width: 100%;
+  height: 10%;
+  margin-top: 10px;
+  margin-bottom: 10px;
+`;
+const ColorBox = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: flex-start;
+  gap: 5px;
+  span {
+    width: 50px;
+    height: 20px;
+    border-radius: 10px;
+    padding: 5px;
+    text-align: center;
+    font-size: 0.75rem;
+    color: white;
+  }
+  @media (max-width: 768px) {
+    span {
+      width: 35px;
+      height: 15px;
+      font-size: 0.5rem;
+
+    }
+  }
+  .today {
+    background-color: #FF8200;
+  }
+  .great {
+    background-color: #32AAFF;
+  }
+  .sorry {
+    background-color: #FF5A5A;
+  }
```

Comment:
지금 코드도 충분히 직관적이고 간결하지만 ColorBox 내부의 클래스들의 색상 값과 텍스트를 상수로 분리하면 유지 보수성을 조금 더 높일 수 있을 것 같아요!

```suggestion
const colorState = {
  today: '#FF8200',
  great: '#32AAFF',
  sorry: '#FF5A5A',
};
```